### PR TITLE
Fix for empty alias

### PR DESCRIPTION
**Pull request by lme-nca**
**Description**

This is an additional fix for issue: 7582 
There was an edge case where the alias array coming from dependency-track is empty. This ensures we correctly fall back to the previous default in this case.

**Test results**

The test now includes this edge case for 1 finding

**Documentation**

-

**Checklist**

This checklist is for your information.

- [ ] Make sure to rebase your PR against the very latest `dev`.
- [ ] Features/Changes should be submitted against the `dev`.
- [ x] Bugfixes should be submitted against the `bugfix` branch.
- [ x] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [ ] Your code is flake8 compliant.
- [ x] Your code is python 3.11 compliant.
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [ ] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ x] Add applicable tests to the unit tests.
- [ ] Add the proper label to categorize your PR.

**Extra information**

Please clear everything below when submitting your pull request, it's here purely for your information.

Moderators: Labels currently accepted for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- performance
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies
- New Migration (when the PR introduces a DB migration)
- settings_changes (when the PR introduces changes or new settings in settings.dist.py)

# Contributors: Git Tips
## Rebase on dev branch
If the dev branch has changed since you started working on it, please rebase your work after the current dev.

On your working branch `mybranch`:
```
git rebase dev mybranch
```
In case of conflict:
```
 git mergetool
 git rebase --continue
 ```

When everything's fine on your local branch, force push to your `myOrigin` remote:
```
git push myOrigin --force-with-lease
```

To cancel everything:
```
git rebase --abort
```

## Squashing commits
```
git rebase -i origin/dev
```
- Replace `pick` by `fixup` on the commits you want squashed out
- Replace `pick` by `reword` on the first commit if you want to change the commit message
- Save the file and quit your editor

Force push to your `myOrigin` remote:
```
git push myOrigin --force-with-lease
```

### CODE REVIEW

File: dojo/tools/dependency_track/parser.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def _convert_dependency_track_finding_to_dojo_finding(self, dependency_track_fin
         # We should collect all the vulnerability ids, the FPF format can add additional IDs as aliases
         # we add these aliases in the vulnerability_id list making sure duplicate findings get correctly deduplicated
         # older version of Dependency-track might not include these field therefore lets check first
-        if 'aliases' in dependency_track_finding['vulnerability']:
+        if 'aliases' in dependency_track_finding['vulnerability'] and len(dependency_track_finding['vulnerability']['aliases']) != 0:
```
Comment (damiencarol) [inline dojo/tools/dependency_track/parser.py:147]: There is a better format for this check:
```suggestion
        if dependency_track_finding['vulnerability'].get('aliases'):
```
It check both null or empty.

File: unittests/scans/dependency_track/many_findings_with_alias.json [hunk 2]
```diff
@@ -585,21 +585,11 @@
...
-                     "NVD": "CVE-2022-2053",
-                     "GITHUB": "GHSA-95rf-557x-44g5"
-                  },
-                  "cveId": "CVE-2022-2053",
-                  "ghsaId": "GHSA-95rf-557x-44g5",
-                  "id": 6883
-               }
-            ],
+            "vulnId": "CVE-2022-2053",
+            "aliases": [],
```
Comment (damiencarol) [inline unittests/scans/dependency_track/many_findings_with_alias.json:589]: You should not modify existing unit tests but create a new one for this bug.

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: unittests/scans/dependency_track/many_findings_with_alias.json (hunk 2 above)
Comment (lme-nca) [reply unittests/scans/dependency_track/many_findings_with_alias.json]: fair enough, created a new test for this

**Review by damiencarol — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]
